### PR TITLE
32 enum for sprites

### LINKED ISSUE #32: Create enum for sprites
Create an enum that holds data related to TiledMapTile.
This means that `Entity.getTile()` and `MapHandler.getBoardLayerTile(int,int)` should return this enum.

It should be the common, asset independent, way of refering to tiles and how they react to the environment.

### PR DESCRIPTION
**Pull request by runalmaas**
fixes #32

### CODE REVIEW

File: src/main/java/no/uib/inf112/core/map/TileType.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    CONVEYOR_FROM_LEFT_UP_DOUBLE_STEP(23, Group.CONVEYOR),
+    CONVEYOR_FROM_BOTTOM_LEFT_DOUBLE_STEP(16, Group.CONVEYOR),
+    CONVEYOR_FROM_BOTTOM_RIGHT_DOUBLE_STEP(17, Group.CONVEYOR),
+    CONVEYOR_FROM_LEFT_DOWN_DOUBLE_STEP(18, Group.CONVEYOR),
+    CONVEYOR_FROM_TOP_LEFT_DOUBLE_STEP(25, Group.CONVEYOR),
+    CONVEYOR_FROM_RIGHT_UP_DOUBLE_STEP(24, Group.CONVEYOR);
+
+
+    private final int id;
+    private final Group group;
```
Comment (kapir) [inline src/main/java/no/uib/inf112/core/map/TileType.java]: This either needs a get or to be public

File: src/main/java/no/uib/inf112/core/player/Robot.java [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
 
-    private final TiledMapTile tile;
     private Direction direction;
     private int x, y;
 
     /**
      * @param x         The x position the player starts at
      * @param y         The y position the player starts at
      * @param direction What direction the player is facing on start
-     * @param headless  true if you want player without graphics (e.g. for testing purposes), false otherwise
```
Comment (kapir) [inline src/main/java/no/uib/inf112/core/player/Robot.java]: why remove descriptive javadoc?

File: src/main/java/no/uib/inf112/core/player/Robot.java [hunk 3]
```diff
@@ -107,8 +99,8 @@ public void move(Movement movement) {
      * Move the robot with given delta to new coordinates
      */
     private void move(int deltaX, int deltaY) {
-        this.x = x + deltaX;
-        this.y = y + deltaY;
+        x = x + deltaX;
+        y = y + deltaY;
     }
```
Comment (kapir) [inline src/main/java/no/uib/inf112/core/player/Robot.java]: not super fan of this change, is it better to do this.x += deltax, or just revert to old notation

File: src/main/java/no/uib/inf112/core/player/Robot.java (hunk 2 above)
Comment (runalmaas) [reply src/main/java/no/uib/inf112/core/player/Robot.java]: @kh498

**Pull request merged**